lib/commons/aria/get-role.js

```diff
@@ -28,6 +29,11 @@ aria.getRole = function getRole(
 		}
 		return aria.isValidRole(role, { allowAbstract: abstracts });
 	});
+
+	if (segments) {
+		return validRoles;
+	}
```

Comment:
It's not at all obvious that `segments` wouldn't return the implicit role. I'm also not a fan of having this turn into a thing that can return either strings or arrays. You're trying to do two things with the same function. I suggest you keep the `getRoleSegments` function around, and have `getRole` call it to get the validRoles value instead.